### PR TITLE
Nested classes shall not be inherited

### PR DESCRIPTION
**Pull request by JaroslavTulach**
Generating the API snapshot and comparing it immediately against itself may lead to 
```
Missing Nested Classes
----------------------
x.P$L_A:            nested public abstract interface static x.P$LB
```
in case of slightly complicated hierarchy of interfaces:
```
            package x;
            public interface P {
                    interface LB extends PI.PJ {
                    }
                    interface A extends P {
                    }
                    interface L extends LB {
                    }
                    class L_A implements L, A {
                    }
            }
            interface PI {
                    interface PJ extends P {
                    }
            }
```

### CODE REVIEW

File: src/main/java/com/sun/tdk/signaturetest/core/MemberCollectionBuilder.java [hunk 1]
```diff
@@ -462,7 +462,7 @@ private void addInheritedFromInterfaces(ClassDescription cl,
                     } else if (memb.isInner()) {
 
                         if (!internalClasses.contains(memb.getName()) && retVal.findSimilar(memb) == null) {
-                            retVal.addMember(memb);
```
Comment (JaroslavTulach) [inline src/main/java/com/sun/tdk/signaturetest/core/MemberCollectionBuilder.java:465]: This is the fix for problems found during [PR-2960](https://github.com/oracle/graal/pull/2960). Is there any reason why "nested classes" list shall be inherited? CCing: @jlahoda, @gradinac

File: src/test/java/org/netbeans/apitest/APITest.java [hunk 2]
```diff
@@ -848,15 +848,39 @@ public void testStrictCheckDiscoversAnAPIChange() throws Exception {
...
         } catch (ExecuteUtils.ExecutionError err) {
             // ok
         }
     }
-    
+
+    public void testStrictNestedInterfaces() throws Exception {
+        String c1 = """
+            package x;
+            public interface P {
```
Comment (JaroslavTulach) [inline src/test/java/org/netbeans/apitest/APITest.java:863]: API that mimics inheritance introduced in [PR-2960](https://github.com/oracle/graal/pull/2960)

Comment (JaroslavTulach) [reply src/test/java/org/netbeans/apitest/APITest.java:863]: If you generate a `.sigfile` for the `P` class, then it looks like this:
```
#Signature file v4.1
#Version 

CLSS public java.lang.Object
cons public init()
meth protected java.lang.Object clone() throws java.lang.CloneNotSupportedException
meth protected void finalize() throws java.lang.Throwable
meth public boolean equals(java.lang.Object)
meth public final java.lang.Class<?> getClass()
meth public final void notify()
meth public final void notifyAll()
meth public final void wait() throws java.lang.InterruptedException
meth public final void wait(long) throws java.lang.InterruptedException
meth public final void wait(long,int) throws java.lang.InterruptedException
meth public int hashCode()
meth public java.lang.String toString()

CLSS public abstract interface x.P
innr public abstract interface static A
innr public abstract interface static L
innr public abstract interface static LB
innr public static L_A

CLSS public abstract interface static x.P$A
 outer x.P
intf x.P

CLSS public abstract interface static x.P$L
 outer x.P
intf x.P$LB

CLSS public abstract interface static x.P$LB
 outer x.P
intf x.P

CLSS public static x.P$L_A
 outer x.P
cons public init()
intf x.P$A
intf x.P$L
supr java.lang.Object
```
There are just four inner classes `A`, `L`, `LB` and `L_A`. Now, when you run a "strictcheck" against that `sigtest` file it reports errors that 
```
Missing Nested Classes
----------------------
x.P$L_A:            nested public abstract interface static x.P$LB
```
However `LB` isn't a nested class of `A`, right!?

The strict checks "inherits" the nested class from `P`. I think that is not useful for anything and hence the commit 
218b7be  disables that.

File: src/main/java/com/sun/tdk/signaturetest/core/MemberCollectionBuilder.java (hunk 1 above)
Comment (gradinac) [reply src/main/java/com/sun/tdk/signaturetest/core/MemberCollectionBuilder.java:465]: Thank you very much for fixing this on such a short notice, we really appreciate it! :)

**Pull request merged**